API.php

```diff
@@ -1224,8 +1224,13 @@ public function disablePreviewMode($idSite, $idContainer)
      * @param string $idContainer  The id of a container, for example "6OMh6taM"
      * @param string $url  The url to enable debug
      */
-    public function changeDebugUrl($idSite, $idContainer, $url)
+    public function changeDebugUrl($idSite, $url)
     {
+        $this->accessValidator->checkWriteCapability($idSite);
+        if (!filter_var($url, FILTER_VALIDATE_URL) || stripos($url, 'http') !== 0) {
```

Comment:
fyi @AltamashShaikh does it make sense to also look at `UrlHelper::isLookLikeSafeUrl()` here additionally? it includes a few additional checks. There's also `isLookLikeUrl()` which may be interesting too not sure.

Reply:
@tsteur Ideally the 'stripos($url, 'http') === 0' should fix most of the cases but adding both the methods makes no harm.

Reply:
👍 that's what I thought. Better be safe. Also in the future someone might add more changes to core and we won't remember to add them here as well meaning then we're covered.